File: week_2/etl_2.py
def etl2(data):
    output = {}

    for key, values in data.items():
        for value in values:
            output[value.lower()] = key

    return output


input = {
    1: ["A", "E"],
    2: ["D", "G"]
}

print(etl2(input))

# expected output:
# {
# 'a': 1,
# 'd': 2,
# 'e': 1,
# 'g': 2
# }

File: week_2/etl_2.js
function etl2(data) {
    const output = {};

    for (const [key, values] of Object.entries(data)) {
        for (const value of values) {
            console.log(value)
            output[value.toLowerCase()] = key;
        }
    }

    return output;

}


input = {
    1: ["A", "E"],
    2: ["D", "G"]
}

console.log(etl2(input))

// # expected output:
// # {
// # 'a': 1,
// # 'd': 2,
// # 'e': 1,
// # 'g': 2
// # }